### PR TITLE
bluetooth: nus: Add option for NUS authentication

### PR DESCRIPTION
**Pull request by hubpav**
This commit adds Kconfig option  CONFIG_BT_NUS_AUTHEN
for authenticated access to Nordic UART service Bluetooth
characteristics. It is disabled by default and has to be explicitly
enabled.

The implementation was inspired by the
CONFIG_MCUMGR_SMP_BT_AUTHEN option.

Signed-off-by: Pavel Hübner <pavel.hubner@hardwario.com>

### CODE REVIEW

File: subsys/bluetooth/services/Kconfig.nus [hunk 1]
```diff
@@ -11,6 +11,11 @@ config BT_NUS
 	  Enable Nordic UART service.
 if BT_NUS
 
+config BT_NUS_AUTHEN
+	bool "Enable authentication for Nordic UART service"
```
Comment (alwa-nordic) [inline subsys/bluetooth/services/Kconfig.nus:15]:
```suggestion
	bool "Require authentication for Nordic UART service"
```

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/services/Kconfig.nus (hunk 1 above)
Comment (hubpav) [reply subsys/bluetooth/services/Kconfig.nus]: Request applied.

**Review by alwa-nordic — APPROVED** [review]

File: subsys/bluetooth/services/Kconfig.nus [hunk 2]
```diff
@@ -11,6 +11,11 @@ config BT_NUS
 	  Enable Nordic UART service.
 if BT_NUS
 
+config BT_NUS_AUTHEN
+	bool "Require authentication for Nordic UART service"
+	help
```
Comment (KAGA164) [inline subsys/bluetooth/services/Kconfig.nus:16]: Consider adding `depends on BT_SMP`

File: subsys/bluetooth/services/Kconfig.nus [hunk 3]
```diff
@@ -11,6 +11,11 @@ config BT_NUS
 	  Enable Nordic UART service.
 if BT_NUS
 
+config BT_NUS_AUTHEN
+	bool "Require authentication for Nordic UART service"
```
Comment (KAGA164) [inline subsys/bluetooth/services/Kconfig.nus:15]: It is nice to keep prompt short. Maybe just "Require authentication"

**Review by KAGA164 — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/services/Kconfig.nus (hunk 3 above)
Comment (hubpav) [reply subsys/bluetooth/services/Kconfig.nus]: @KAGA164 Okay, I spent a little time looking around and deleted the help text, while using the prompt [from this option](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/index.html#CONFIG_BT_HRS_DEFAULT_PERM_RW_AUTHEN). It best describes what it does and makes it consistent in the tree. Do you agree?

Comment (KAGA164) [reply subsys/bluetooth/services/Kconfig.nus]: The prompt test is good now, but please bring back the help text to render this nice in the documentation. If you want to do it in consistent way with other services you can use choice in Kconfig like it is done in the HRS https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/Kconfig.hrs#L11
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/hrs.c#L46.
If you decide to go with approach from HRS, you can also add option for access with encryption

Comment (hubpav) [reply subsys/bluetooth/services/Kconfig.nus]: @KAGA164 Okay. Let's keep it simple for now - I have returned back the help text. But we can consider the choice option in the future. The HRS sample gives the most flexibility.

**Review by KAGA164 — APPROVED** [review]

**Comment by KAGA164** [comment]
@hubpav Please rebase on current main, it should make CI green and then I will put this PR in a merge queue

**Comment by hubpav** [comment]
> @hubpav Please rebase on current main, it should make CI green and then I will put this PR in a merge queue

Done.

**Pull request merged**